db/db_impl/db_impl_open.cc

```diff
@@ -1051,19 +1053,28 @@ Status DBImpl::RecoverLogFiles(const std::vector<uint64_t>& log_numbers,
       if (flushed || cfd->mem()->GetFirstSequenceNumber() == 0) {
         edit->SetLogNumber(max_log_number + 1);
       }
+    }
+    if (status.ok()) {
       // we must mark the next log number as used, even though it's
       // not actually used. that is because VersionSet assumes
       // VersionSet::next_file_number_ always to be strictly greater than any
       // log number
       versions_->MarkFileNumberUsed(max_log_number + 1);
-      status = versions_->LogAndApply(cfd, *cfd->GetLatestMutableCFOptions(),
-                                      edit, &mutex_);
-      TEST_SYNC_POINT_CALLBACK("DBImpl::RecoverLogFiles:AfterLogAndApply",
```

Comment:
The new crash location is good. But I still think it is valuable to test crashing in this for CF loop, e.g. somewhere like line 1015? Can we preserve that too?

Reply:
The `for` loop does not modify MANIFEST. The bug we are fixing here is related to the all-or-nothing property of MANIFEST.
If we crash in the `for` loop, no new MANIFEST will be written.

Reply:
Done